Split --uktdd-body-position-try-fallbacks into two fixed slots (#404)
acusti/uikit: .changeset/dropdown-split-fallback-slots.md

---

> '@acusti/dropdown': minor

Author: Agreed — bumped to `major` in 1f247e58. It's a breaking rename of public custom properties, and it matches the repo's convention (e.g. `dropdown-rename-position-try-fallbacks` is major for renaming this same property). The dropdown alpha train is already headed to a major release, so this just categorizes the change correctly under Major Changes.

---

Replace the `--uktdd-body-position-try-fallbacks` list with two fixed
slots, `--uktdd-body-position-try-fallback-1` and
`--uktdd-body-position-try-fallback-2`, and rename the submenu’s list to

> the singular `--uktdd-submenu-position-try-fallback`

Author: Good catch — reconciled in 1f247e58. Updated the sibling changesets that referenced the old list-form names to the slot names so the combined release notes are consistent: `dropdown-fill-fallbacks-within-evaluation-limit`, `dropdown-submenu-fills-beside-parent`, `dropdown-rename-position-try-fallbacks` (its "after" example is now the two-slot form), and `nested-dropdown-submenus` (→ singular `--uktdd-submenu-position-try-fallback`).

One I left for the maintainer: `dropdown-prefer-authored-direction` still references the old names *and* describes the superseded "four appended `--uktdd-fill` fallbacks" design (this session's fill work replaced those four tactic variants with two named fills). That's a design-narrative change beyond a name swap, so it wants an editorial decision rather than a silent rewrite.

> the singular `--uktdd-submenu-position-try-fallback`

Author: Update: reconciled `dropdown-prefer-authored-direction` too (0c7ed9ff), and reframed the fill narrative across the train. Since the whole fill system churned within this one unreleased alpha — `prefer-authored-direction` introduced four `--uktdd-fill` tactic fallbacks, the evaluation-limit change then replaced them with two named fills — describing "added four, then replaced with two" only muddied the notes. Now `prefer-authored-direction` keeps its actual contribution (authored direction wins over most-height, generic fill-and-scroll last resort) with no four-fill mechanics, and the evaluation-limit changeset describes the two named fills as the design rather than a replacement. The combined changelog now reads as one coherent final-state story.

The author fallbacks sit between the primary placement and the appended
fills, and the component’s budget is exactly two of them (two author

> fallbacks + two fills = four past the base = the spec’s guaranteed

Author: ```suggestion
fallbacks + two fills = four, plus the base = the spec’s guaranteed
```

Author: Applied in 38351e75.

five-option floor). As a comma-separated list, that limit was only

> documented, and a third entry silently pushed a fill past the floor —

Author: ```suggestion
documented, and a third entry silently pushed a fill past the floor,
```

Author: Applied in 38351e75.

letting a trigger near a viewport edge open off-screen. Splitting the list
into two named slots makes the two-option budget structural: there is no
third slot to overflow into. The submenu, which affords a single author
fallback, likewise becomes a single named slot.

Migration: set `--uktdd-body-position-try-fallback-1` and `-2` where you
previously set the two entries of `--uktdd-body-position-try-fallbacks`
(the direction-recipe table in the README lists the per-direction values),
and `--uktdd-submenu-position-try-fallback` (singular) where you set the
submenu list.

Also adds a shipped no-op placement, `--uktdd-noop`, for a recipe with a
single meaningful fallback (a centered menu, say): set slot 1 and leave
slot 2 as `--uktdd-noop` so it doesn’t inherit the default’s flip. It
applies no overrides, so it evaluates as the primary placement — already

> rejected by the time it’s reached — and is skipped.

Author: nit:
```suggestion
applies no overrides, so it evaluates as the primary placement and is skipped.
```

Author: Applied in 96283213.
